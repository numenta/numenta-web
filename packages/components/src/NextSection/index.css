/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.nextSection {
  composes: mt4 from 'tachyons';
}
