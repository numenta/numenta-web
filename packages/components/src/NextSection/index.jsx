// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import IconArrow from 'react-icons/lib/fa/caret-right'
import React from 'react'

import Strong from '../Strong'
import TextLink from '../TextLink'

import styles from './index.css'


/**
 *
 */
const NextSection = ({title, url}) => (
  <div className={styles.nextSection}>
    <Strong>Next:</Strong> {' '}
    <TextLink to={url}>
      {title} <IconArrow />
    </TextLink>
  </div>
)

NextSection.propTypes = {
  title: React.PropTypes.string.isRequired,
  url: React.PropTypes.string.isRequired,
}

export default NextSection
