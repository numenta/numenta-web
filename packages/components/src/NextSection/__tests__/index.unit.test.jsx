import React from 'react'
import renderer from 'react-test-renderer'

import NextSection from '../../NextSection'


describe('NextSection React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <NextSection
        title="Mission & History"
        url="/mission-and-history/"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
