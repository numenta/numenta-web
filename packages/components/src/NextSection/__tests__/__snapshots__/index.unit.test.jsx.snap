exports[`NextSection React component Renders correctly 1`] = `
<div
  className="nextSection">
  <strong
    className="strong">
    Next:
  </strong>
   
   
  <a
    className="textlink"
    onClick={[Function]}
    style={Object {}}
    target={undefined}>
    Mission & History
     
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em">
      <g>
        <path
          d="m26.4 20q0 0.6-0.5 1l-10 10q-0.4 0.4-1 0.4t-1-0.4-0.4-1v-20q0-0.6 0.4-1t1-0.4 1 0.4l10 10q0.5 0.4 0.5 1z" />
      </g>
    </svg>
  </a>
</div>
`;
